### DISCUSSION TITLE
Can I make a branchName that doesn't contain baseBranch?

### DISCUSSION THREAD

**Discussion by justinp** [renovatebot/renovate#22589] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.109.0

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Our CI system is configured to recognize multiple active develop branches (e.g., `develop/1.3.2` and `develop/2.1.0`) each with multiple feature branches (e.g., `feature/1.3.2/JIRA-223`). The build scripts use these naming patterns to derive artifact revisions.

I'm trying to make it so that renovate checks each of the develop branches for updates and, if found, creates feature branches with the proposed updates (to work with our existing CI setup). I can't however make it so that the created branch name doesn't contain the baseBranch name and due to our naming convention, that's not acceptable.

### What you tried so far.

I tried changing `branchName`, `branchPrefix`, `additionalBranchPrefix` and `branchTopic`. There doesn't seem to be anything I can do to completely generate the branch name without it just being prefixes and suffixes on the base branch name. The documentation seems to confirm this as well.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
When baseBranches are in use, the base branch is used to distinguish between each of them.  Doing it that way is obviously a good way to ensure uniqueness between base branches.

Could you give an example of a branch name which is not allowed by your rules, and an alternative (with uniqueness) which would have? e.g. would replacing `/` with `-` be enough?

**Comment by justinp** [comment]
Hey, thanks for the response!

What I'm looking to do is something like:
* `develop/1.3.2` -> `feature/1.3.2/renovate-org.scalamock-scalamock-5.x`
* `develop/2.1.0` -> `feature/2.1.0/renovate-org.scalamock-scalamock-5.x`

So, I want to use my knowledge of the format of the base branch name to extract the unique part and then use that to generate unique feature branch names. The tricky part here seems to be getting rid of the `develop/`.

`feature/develop/1.3.2/renovate-org.scalamock-scalamock-5.x` is _not_ allowed by our build scripts because it's looking for the pattern `feature/X.Y.Z/name` or `develop/X.Y.Z` to determine the artifact revision.
